### DISCUSSION TITLE
p77 リスト8.3のとリスト8.4について

### DISCUSSION THREAD

**Discussion by katutoshi** [budougumi0617/go_todo_app#42] [Ideas]
p77についてです。
### リスト8.3
エラーを返した関数名を埋め込みたいとすると、func GetAuthorNameで返すエラー は `fmt.Errorf("GetAuthor: %v", err)` じゃなくて `fmt.Errorf("GetAuthorName: %v", err)` が正しそうです

### リスト8.4
リスト8.3を前提とすると
`GetBookSummary: GetAuthorName: GetUser: id is invalid` は `GetBookSummary: GetAuthorName: GetAuthor: id is invalid`
が正しそうです

**Comment by fuuukeee3** [comment]
@katutoshi さんのご指摘に相乗りですいません。
P77 リスト8.3の3行目
`return errors.New("GetAuthor: id is invalid")`となっていますが、`return nil, errors.New("GetAuthor: id is invalid")`として一つ目の返り値で`nil`を返す必要があるのではないでしょうか。

**Comment by budougumi0617** [comment]
ご指摘ありがとうございます！今週中に確認・正誤表の更新をさせていただきます！とりいそぎお礼まで…

**Comment by budougumi0617** [comment]
ご指摘ありがとうございました。正誤表に反映しました。
- https://github.com/budougumi0617/go_todo_app/pull/46

// 今いただいている他の未対応の指摘もひととおり対応しだいリリースタグつけようと思います 🙇

**Reply by budougumi0617** [reply]
正誤表を更新しリリースタグをつけました。ご報告ありがとうございました。
https://github.com/budougumi0617/go_todo_app/releases/tag/v1.0.5
https://github.com/budougumi0617/go_todo_app/blob/v1.0.5/errata.md
